### PR TITLE
[1.3.5-prepare][Fix-3298][K8s]Fix task log disappear after recreating or upgrading helm release

### LINKED ISSUE #3298: [Bug][k8s]when recreate or upgrade helm release, the host ip changed
**Describe the bug**
When DS was deployed to k8s by helm, if you upgrade the release or recreate the release, the ip of worker server and other server will be changed, so when you try to query the log of history task, you will get empty log.

**To Reproduce**
Steps to reproduce the behavior, for example:
1. deploy the cluster use helm
2. recreate the deployed release or just upgrade
3. query a task which runned before upgrade/recreate
4. no log will show

**Expected behavior**
log should be show

**Screenshots**

image

**Which version of Dolphin Scheduler:**
 -[1.3.1-release]

**Additional context**

**Requirement or improvement**

### PR DESCRIPTION
**Pull request by chengshiwen**
## *Tips*
- *Thanks very much for contributing to Apache DolphinScheduler.*
- *Please review https://dolphinscheduler.apache.org/en-us/community/index.html before opening a pull request.*

## What is the purpose of the pull request

**Fix task log disappear after recreating or upgrading helm release**

This closes #3298

## Brief change log

  - *Alter column host varchar(15) to varchar(135) for long host*
  - *Support to get hostname under kubernetes*
  - *Fix task log disappear after recreating or upgrading helm release*

## Verify this pull request

*(Please pick either of the following options)*

This change added tests and can be verified as follows:

*(example:)*

  - *Manually verified the change by testing locally.*

Video check: https://github.com/chengshiwen/dolphinscheduler-test/blob/master/fix-k8s-3298.mp4?raw=true

### CODE REVIEW

File: dolphinscheduler-server/src/test/java/org/apache/dolphinscheduler/server/master/registry/MasterRegistryTest.java [hunk 1]
```diff
@@ -55,7 +56,7 @@ public void testRegistry() throws InterruptedException {
         masterRegistry.registry();
         String masterPath = zookeeperRegistryCenter.getMasterPath();
         TimeUnit.SECONDS.sleep(masterConfig.getMasterHeartbeatInterval() + 2); //wait heartbeat info write into zk node
-        String masterNodePath = masterPath + "/" + (Constants.LOCAL_ADDRESS + ":" + masterConfig.getListenPort());
+        String masterNodePath = masterPath + "/" + (OSUtils.getAddr(Constants.LOCAL_ADDRESS, masterConfig.getListenPort()));
```
Comment (lenboo) [inline dolphinscheduler-server/src/test/java/org/apache/dolphinscheduler/server/master/registry/MasterRegistryTest.java:59]: the "/"  can be replaced to  constant variable.

**Review by davidzollo — APPROVED** [review]
+1
good job

**Comment by davidzollo** [comment]
good job
don't forget to submit this PR  to dev

**Comment by chengshiwen** [comment]
@dailidong Got it

File: dolphinscheduler-server/src/test/java/org/apache/dolphinscheduler/server/master/registry/MasterRegistryTest.java (hunk 1 above)
Comment (chengshiwen) [reply dolphinscheduler-server/src/test/java/org/apache/dolphinscheduler/server/master/registry/MasterRegistryTest.java:59]: Resolved by #4704

**Pull request merged**